### DISCUSSION TITLE
firebase_auth starts throwing MissingPluginException when Flutter Web app is hosted on Firebase

### DISCUSSION THREAD

**Discussion by zoborowi** [firebase/flutterfire#10109] [Q&A]
Firebase authentication is working fine local when I run a Flutter Web app in Chrome but once I publish and host it in Firebase I start getting uncaught exceptions:

Uncaught MissingPluginException(No implementation found for method Auth#registerIdTokenListener on channel plugins.flutter.io/firebase_auth)
Uncaught MissingPluginException(No implementation found for method Auth#registerAuthStateListener on channel plugins.flutter.io/firebase_auth)

Thoughts?

**Comment by zoborowi** [comment]
If I use `flutter build web --web-renderer html` instead of just `flutter build web` and then deploy I don't have issues with MissingPluginExceptions. I am not sure why. If someone has an explanation of why this is so I would love to understand.

**Reply by ringonotts** [reply]
Thank you so much for posting this solution as I have just faced the same issue - and you solved it!
I'm glad you solved it because no one from google seems to be helping you....
Ringo

**Reply by rossnguyen** [reply]
Your solution works very well, many thanks.

**Reply by hjvanranden-topicus** [reply]
Solved it for me too, thanks!

**Reply by yasfdany** [reply]
solved on firebase_storage too, thanks

**Reply by orgesazeqiri** [reply]
It solved this issue for me as well but now running it without the rendering option it shows different font style and it does not render some svgs correctly. Does anyone know the solution to this?

**Reply by kfieldi5** [reply]
This did not solve it for me sadly, I have always been running the `-web-renderer html` flag since I have 3D stuff going on via model_viewer_plus and need it.

Does anyone know the cause?
